src/blockencodings.cpp

```diff
@@ -43,6 +44,32 @@ uint64_t CBlockHeaderAndShortTxIDs::GetShortID(const uint256& txhash) const {
     return SipHashUint256(shorttxidk0, shorttxidk1, txhash) & 0xffffffffffffL;
 }
 
+namespace {
+    struct ShortIdIndexPair {
+        uint64_t shortid : 48;
+        uint16_t index : 16;
```

Comment:
in order to make these packed together, you should change ``uint16_t`` to ``uint64_t``. Otherwise these are two separate bitfields with different allocation units.